src/tests/Common/XUnitWrapperLibrary/TestFilter.cs

```diff
@@ -88,21 +89,49 @@ public NotClause(ISearchClause inner)
         public bool IsMatch(string fullyQualifiedName, string displayName, string[] traits) => !_inner.IsMatch(fullyQualifiedName, displayName, traits);
     }
 
-    private ISearchClause? _filter;
+    private readonly ISearchClause? _filter;
+    private readonly HashSet<string>? _testExclusionList;
```

Comment:
Lets add a comment here about why we are supporting a separate test exclusion list model.

cc: @agocke, maybe we can reuse this model for the libraries tests if we extend this generator to be used repo-wide?

Reply:
Comment added in 2nd iteration, happy to iterate on its wording.

Reply:
Frankly speaking I'm somewhat unsure whether the existing system of <code>TestFilter.AndClause</code> vs. <code>TestFilter.OrClause</code> is not a bit of an overshoot - perhaps an "allow-list" and a "block-list" could serve the same purpose and not suffer parenthesis placement issues that occur in arbitrary orderings of And vs. Or clause compositions.

Reply:
The goal of the `AndClause` and `OrClause` is to match exactly how `dotnet test`'s `--filter` argument parses the expression (as soon as I have time to implement it). I want to support the exact same filtering mechanism that `dotnet test` supports for consistency throughout the repo.